src/tools/copyset_check_core.cpp

```diff
@@ -154,12 +154,14 @@ ChunkServerHealthStatus CopysetCheckCore::CheckCopysetsOnChunkServer(
     bool isHealthy = true;
     butil::IOBuf iobuf;
     int res = QueryChunkServer(chunkserverAddr, &iobuf);
+    statisticsMutest.lock();
```

Comment:
This lock will bring some difficulties to maintenance.
The real problem that causes a lot of times is sending RPC to chunkserver, so I think you can change schemes here. You can concurrently send RPC to chunkserver and save RPC response. And if all RPC returned or failed, gather all responses and process in a single thread. So, this will not bring data race to processing RPC responses.

Reply:
> This lock will bring some difficulties to maintenance.
> The real problem that causes a lot of times is sending RPC to chunkserver, so I think you can change schemes here. You can concurrently send RPC to chunkserver and save RPC response. And if all RPC returned or failed, gather all responses and process in a single thread. So, this will not bring data race to processing RPC responses.

I agree this suggestion. I think it's better handle RPC request by concurrent and handle the RPC responses together. It will make the logic more clearly and you can reuse a lot of code.

Reply:
> This lock will bring some difficulties to maintenance.
> The real problem that causes a lot of times is sending RPC to chunkserver, so I think you can change schemes here. You can concurrently send RPC to chunkserver and save RPC response. And if all RPC returned or failed, gather all responses and process in a single thread. So, this will not bring data race to processing RPC responses.

BTW：You only need one lock to save RPC responses in parallel.

Reply:
fixed